clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsResponse.java

```diff
@@ -313,10 +313,12 @@ private static FinalizedFeatureKeyCollection createFinalizedFeatureKeys(
         for (Map.Entry<String, Short> feature : finalizedFeatures.entrySet()) {
             final FinalizedFeatureKey key = new FinalizedFeatureKey();
             final short versionLevel = feature.getValue();
-            key.setName(feature.getKey());
-            key.setMinVersionLevel(versionLevel);
-            key.setMaxVersionLevel(versionLevel);
-            converted.add(key);
+                if (versionLevel != 0) {
```

Comment:
The indentation seems off. I think you need to remove 4 spaces.

Let's write comment explaining why we only do this when the version level is not 0?